File: guides/voice/conference-calls-guide/moderated-conference/moderated-conference.5.x.py
from flask import Flask, request
from twilio import twiml

app = Flask(__name__)

# Update with your own phone number in E.164 format
MODERATOR = '+15558675310'


@app.route("/voice", methods=['GET', 'POST'])
def call():
    """Returns TwiML for a moderated conference call"""
    # Start our TwiML response
    response = twiml.Response()

    # Start with a <Dial> verb
    with response.dial() as dial:
        # If the caller is our MODERATOR, then start the conference when they
        # join and end the conference when they leave
        if request.values.get('From') == MODERATOR:
            dial.conference(
                'My conference',
                startConferenceOnEnter=True,
                endConferenceOnExit=True)
        else:
            # Otherwise have the caller join as a regular participant
            dial.conference('My conference', startConferenceOnEnter=False)

    return str(response)

if __name__ == "__main__":
    app.run(debug=True)

File: guides/voice/conference-calls-guide/moderated-conference/moderated-conference.5.x.js
const express = require('express');
const VoiceResponse = require('twilio').twiml.VoiceResponse;
const urlencoded = require('body-parser').urlencoded;

// Update with your own phone number in E.164 format
const MODERATOR = '+15558675310';

const app = express();

// Parse incoming POST params with Express middleware
app.use(urlencoded({ extended: false }));

// Create a route that will handle Twilio webhook requests, sent as an
// HTTP POST to /voice in our application
app.post('/voice', (request, response) => {
  // Use the Twilio Node.js SDK to build an XML response
  const twiml = new VoiceResponse();

  // Start with a <Dial> verb
  const dial = twiml.dial();
  // If the caller is our MODERATOR, then start the conference when they
  // join and end the conference when they leave
  if (request.body.From == MODERATOR) {
    dial.conference('My conference', {
      startConferenceOnEnter: true,
      endConferenceOnExit: true,
    });
  } else {
    // Otherwise have the caller join as a regular participant
    dial.conference('My conference', {
      startConferenceOnEnter: false,
    });
  }

  // Render the response as XML in reply to the webhook request
  response.type('text/xml');
  response.send(twiml.toString());
});

// Create an HTTP server and listen for requests on port 3000
console.log('Twilio Client app HTTP server running at http://127.0.0.1:3000');
app.listen(3000);
